src/AutoConstructor.Generator/AutoConstructorGenerator.cs

```diff
@@ -172,11 +172,16 @@ private static List<FieldInfo> GetFieldsFromSymbol(Compilation compilation, INam
         return symbol.GetMembers().OfType<IFieldSymbol>()
             .Where(x => x.CanBeInjected(compilation)
                 && !x.IsStatic
-                && x.IsReadOnly
+                && (x.IsReadOnly || IsPropertyWithExplicitInjection(x))
                 && !x.IsInitialized()
                 && !x.HasAttribute(Source.IgnoreAttributeFullName, compilation))
             .Select(x => GetFieldInfo(x, compilation, emitNullChecks))
             .ToList();
+
+        bool IsPropertyWithExplicitInjection(IFieldSymbol x)
+        {
+            return x.IsImplicitlyDeclared && x.HasAttribute(Source.InjectAttributeFullName, compilation);
```

Comment:
```suggestion
            return x.AssociatedSymbol is not null && x.HasAttribute(Source.InjectAttributeFullName, compilation);
```
I would prefer something like this, your code will work, but I've used the check on AssociatedSymbol in other places, so it will more coherent.